### ISSUE TITLE
libwinpthread seems to be causing issues when cloing repositories.

### ISSUE THREAD

**Issue by cwgreene** [git-for-windows/git#3444]
- [ ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing
 
 For an large internal repository, git clone fails with the following:
 
 ```
 Cloning into 'localcheckout'...
remote: Enumerating objects: 159058, done.
remote: Counting objects: 100% (159058/159058), done.
remote: Compressing objects: 100% (31922/31922), done.
fetch-pack: unexpected disconnect while reading sideband packet/s
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
```

 
 Copying the libwinpthread-1.dll file from the 2.33.0 release onto the 2.33.2 release prevents this from happening.

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
 64 bit

```
$ git --version --build-options
```
```
git version 2.32.0.windows.2
cpu: x86_64
built from commit: 3d45ac813c4adf97fe3733c1f763ab6617d5add5
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

** insert your machine's response here **
```
Microsoft Windows [Version 10.0.19042.1237]

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt
```
```
Editor Option: VIM
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Core
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Powershell

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

git clone git@internal.git.server:/path/to/repo

```
git clone git@internal.git.server:/path/to/repo
```
 - What did you expect to occur after running these commands?

Cloned repository

 - What actually happened instead?
```
Cloning into 'local_checkout'...
remote: Enumerating objects: 159058, done.
remote: Counting objects: 100% (159058/159058), done.
remote: Compressing objects: 100% (31922/31922), done.
fetch-pack: unexpected disconnect while reading sideband packet/s
fatal: early EOF
fatal: fetch-pack: invalid index-pack
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Internal, cannot link I'm afraid.

**Comment by cwgreene** [comment]
How can I figure out which version of libwinpthread was used in the two versions?

**Comment by rimrul** [comment]
> How can I find out which version of libwinpthread was used in each version? 

[We keep a list of package versions for each release](https://github.com/git-for-windows/build-extra/tree/main/versions)

**Comment by cwgreene** [comment]
> 
> 
> > How can I find out which version of libwinpthread was used in each version?
> 
> [We keep a list of package versions for each release](https://github.com/git-for-windows/build-extra/tree/main/versions)

Awesome! Thank you! I'll dig into that. :)

**Comment by dscho** [comment]
> 
> 
> > > How can I find out which version of libwinpthread was used in each version?
> > 
> > 
> > [We keep a list of package versions for each release](https://github.com/git-for-windows/build-extra/tree/main/versions)
> 
> Awesome! Thank you! I'll dig into that. :)

@cwgreene any news?

**Comment by cwgreene** [comment]
@dscho no unfortunately. I have honestly been having trouble setting up the environment to build the dlls. I think I somehow managed to build the linux target, but not the windows one (as no dlls were created after running `make`). Attempts to build using the windows sdk install environment ran into issues (missing a header). Will be resuming attacking this, my pardon for the delays on more information.

**Comment by dscho** [comment]
I'll just close this until we have further information.

**Comment by jmigual** [comment]
Hello, this issue is still there with git version 2.34.1.windows.1 and the mailing list discussion does not seem solved. Is there anything we can do to solve it?

**Comment by dscho** [comment]
@jmigual seems the conversation is stuck, indeed. In https://lore.kernel.org/git/YXMTuMGWQD4Swwt4@coredump.intra.peff.net/, I see an unanswered question:

> I wonder if it's possible to get some kind of equivalent trace
> for the version of ssh in use here. Even if not strace, I think "ssh
> -vv" details some of this (but do make sure you're not using a
> ControlMaster, as that hides a lot of it in a separate process).

And in https://lore.kernel.org/git/b802e803-520d-49f5-a845-8e3c94f1c0ee@gmail.com/:

> Try adding GIT_TRACE_PACKET=true when attempting to clone the 
> problematic repo.

I guess the good people who responded to the mail thread are _kind of_ waiting for answers...

**Comment by smuehlst** [comment]
I'm running into this occasionally on Windows 11 with `git version 2.36.1.windows.1` when doing a `git pull` or `git clone`. In my experience this happens only when setting up Git for Windows to use the ssh.exe that comes with Windows itself.

Another observation is that setting environment variables to turn on tracing works around the problem. I followed the advice in this StackOverflow discussion:

https://stackoverflow.com/a/67390274/2995726

Setting `core.compression 0` etc. alone didn't prevent the problem from occurring. But with the following environment variables set in PowerShell the same command that reliably failed before suddenly succeeds:

```
$env:GIT_TRACE_PACKET=1
$env:GIT_TRACE=1
$env:GIT_CURL_VERBOSE=1
```

**Comment by lautenbde** [comment]
Seeing this issue, too,  with git version 2.39.0.windows.1 on Win 11. Sometimes clone is working - when you are lucky - but most attempts fail with said error. Seems still no fix for this?
Does it mean you need to stick to version 2.32 in order to avoid this issue?

**Comment by lautenbde** [comment]
Reverted back to git version 2.32.0.windows.2 - issue gone! This issue should be re-opened...

**Comment by dscho** [comment]
@lautenbde how about answering those open questions? That would help drive this ticket forward (in contrast to hitting the Reopen button, which does not help answering those questions one bit).

**Comment by cwgreene** [comment]
@lautenbde can you verify which version of `ssh` you're using?

There has been a long standing bug in windows ssh that I believe was the actual root cause of this issue.

**Comment by cwgreene** [comment]
@dscho I would advocate closing this issue. The root cause for this issue is fixed by this pull request in the windows port of openssh.

https://github.com/PowerShell/openssh-portable/pull/628

**Comment by dscho** [comment]
Thank you for following up, @cwgreene!

**Comment by lautenbde** [comment]
@cwgreene It is version 8.6.0.1 / OpenSSH_8.6p for Windows. But seems the PR you are referring to is still waiting for merge?

**Comment by cwgreene** [comment]
@lautenbde yes, it is still waiting to be merged. Main thing I wanted to confirm was that you were using openssh for windows.

**Comment by j-ulrich** [comment]
Issue seems fixed with [Win32-OpenSSH version v9.2.2.0p1](https://github.com/PowerShell/Win32-OpenSSH/releases/tag/v9.2.2.0p1-Beta)
